File: LeetCode-Problems/Medium/Remove-Node-From-End.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        l = 0
        curr = head
        while (curr != None):
            curr = curr.next
            l = l + 1
            
        if (l == n):
            return head.next
        p = 0
        curr = head
        while (p < (l - n) - 1):
            curr = curr.next
            p = p + 1
        
        curr.next = curr.next.next
        return head

File: LeetCode-Problems/Medium/Remove-Node-From-End.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode curr = head;
        
        while (curr != null){
            curr = curr.next;
            len++;
        }
        
        int pointer = 0;
        curr = head;
        if (len - n == 0){
            return head.next;
        }
        while (pointer < (len - n) - 1){
            curr = curr.next;
            pointer++;
        }
        curr.next = curr.next.next;
        return head;
        
    }
}
